rdflib/term.py

```diff
@@ -599,75 +657,114 @@ class Literal(Identifier):
     _value: Any
     _language: Optional[str]
     # NOTE: _datatype should maybe be of type URIRef, and not optional.
-    _datatype: Optional[URIRef]
+    _datatype: URIRef
     _ill_typed: Optional[bool]
     __slots__ = ("_language", "_datatype", "_value", "_ill_typed")
 
+    # @overload
+    # def __new__(
+    #     cls,
+    #     lexical_or_value: Any,
+    #     *,
+    #     lang: str,
+    #     normalize: Optional[bool] = None,
+    # ) -> Literal:
+    #     ...
+
+    # @overload
+    # def __new__(
+    #     cls,
+    #     lexical_or_value: Any,
+    #     *,
+    #     normalize: Optional[bool] = None,
+    # ) -> Literal:
+    #     ...
+
+    # @overload
+    # def __new__(
+    #     cls,
+    #     lexical_or_value: Any,
+    #     *,
+    #     datatype: str,
+    #     normalize: Optional[bool] = None,
+    # ) -> Literal:
+    #     ...
+
+    # @overload
+    # def __new__(
+    #     cls,
+    #     lexical_or_value: str,
+    #     lang: Optional[str],
+    #     datatype: str,
+    #     normalize: Optional[bool] = None,
+    # ) -> Literal:
+    #     ...
+
     def __new__(
         cls,
         lexical_or_value: Any,
         lang: Optional[str] = None,
         datatype: Optional[str] = None,
         normalize: Optional[bool] = None,
     ) -> "Literal":
-        if lang == "":
-            lang = None  # no empty lang-tags in RDF
-
         normalize = normalize if normalize is not None else rdflib.NORMALIZE_LITERALS
-
-        if lang is not None and datatype is not None:
-            raise TypeError(
-                "A Literal can only have one of lang or datatype, "
-                "per http://www.w3.org/TR/rdf-concepts/#section-Graph-Literal"
-            )
-
-        if lang is not None and not _is_valid_langtag(lang):
-            raise ValueError(f"'{str(lang)}' is not a valid language tag!")
-
         if datatype is not None:
             datatype = URIRef(datatype)
 
+        if lang is not None or datatype == _RDF_LANGSTRING:
+            if lang is None:
+                raise ValueError(f"language cannot be None if datatype is {datatype}")
+            if not lang:
+                raise ValueError("language cannot be empty string")
+            if not _lang_tag_regex.match(lang):
+                raise ValueError(
+                    "language tag must match the pattern "
+                    "^[a-zA-Z]+(?:-[a-zA-Z0-9]+)*$ "
+                    f"but was {lang}"
+                )
+
+            if datatype is None:
+                datatype = _RDF_LANGSTRING
+
```

Comment:
Let `q=Literal("ex", datatype=something)`. This will also overwrite any datatype given to a new Literal `Literal(q)`.
```
#prevous:
Literal(q)
# >> Literal("ex", datatype=something)

# now:
Literal(q)
# >> Literal("ex", datatype=RDF.langString)
```

Before the datatype was inherited. But i havent seen this behaviour documented, so there was now expected behaviour before. But we could mention this change in behaviour nonetheless.
Also this will handle issue #2603